app/models/piece.rb

```diff
@@ -39,6 +39,15 @@ def capturable?(destination_row, destination_column)
     Piece.where(game_id: game.id, row: destination_row, column: destination_column, in_game: true, color: opposite_color).exists?
   end
 
+  def any_move?
+    8.times do |r|
+      8.times do |c|
+        return true if unoccupied_space?(r, c) && valid_move?(r, c)
```

Comment:
Yeah, it should be covered in the valid_move? method so I'm pretty sure you can just take that out.